pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -242,6 +252,23 @@ private void putTableConfig(ZNRecord znRecord)
       _tableNameMap.put(tableNameWithType.toLowerCase(), tableNameWithType);
       _tableNameMap.put(rawTableName.toLowerCase(), rawTableName);
     }
+    addOverrideHintsColumns(tableConfig, schemaName, _caseInsensitive);
+  }
+
+  private void addOverrideHintsColumns(TableConfig tableConfig, String schemaName, boolean caseInsensitive) {
```

Comment:
This can cause problems when the schema is shared by multiple table configs (e.g. hybrid table), and they have different index configured. We should avoid making dependencies between table config and schema. If this is for the column name check, we can probably skip the check for timestamp index column if explicitly called

Reply:
Added the check to ignore virtual columns.